Readability.js

```diff
@@ -1198,62 +1198,66 @@ Readability.prototype = {
     var values = {};
     var metaElements = this._doc.getElementsByTagName("meta");
 
-    // Match "description", or Twitter's "twitter:description" (Cards)
-    // in name attribute.
-    var namePattern = /^\s*((twitter)\s*:\s*)?(description|title)\s*$/i;
+    // property is a space-separated list of values
+    var propertyPattern = /\s*(dc|dcterm|og|twitter)\s*:\s*(author|creator|description|title)\s*/gi;
 
-    // Match Facebook's Open Graph title & description properties.
-    var propertyPattern = /^\s*og\s*:\s*(description|title)\s*$/i;
+    // name is a single value
+    var namePattern = /^\s*(?:(dc|dcterm|og|twitter)\s*[\.:]\s*)?(author|creator|description|title)\s*$/gi;
```

Comment:
Can you remove the `g` flag here? Otherwise, I believe it'll reintroduce issue #480 , see the discussion in the first answer here: https://stackoverflow.com/questions/209732/why-am-i-seeing-inconsistent-javascript-logic-behavior-looping-with-an-alert-v - you're using `namePattern` with `.test()` which will keep the last matching index and won't match the next time you run it.

Reply:
No. We need global matches so we can find all the metadata candidates. It lets the logic find the best data source and use them in a predictable order (Dublin Core first, then Open Graph, then HTML5/native, and lastly Twitter).
https://github.com/mozilla/readability/pull/478/files/5b1dabb1700510e9466e79ea0e03cc8542cbcb8d#diff-06d8d22df421dacde90a2268083424abR1255

Reply:
We need the global modifier. But I think I can fix the original issue here without loosing the g.

The global lets us find every metadata candidate so that we can select from the candidates in a defined order (Dublin Core first followed by Open Graph, HTML5 native, and lastly Twitter). This makes the metadata selection process less random to web developers.

Reply:
But we don't use `namePattern` with `match`, only with `test`, which doesn't find more than 1 candidate, only returns true/false, right?

Reply:
That is, I can see you want the global modifier with `propertyPattern`, but not (in the current implementation) with `namePattern.